components/backup-stream/src/subscription_manager.rs

```diff
@@ -552,22 +568,59 @@ where
     }
 
     async fn start_observe(&self, region: Region) {
+        self.retry_start_observe(region, 0).await
+    }
+
+    async fn retry_start_observe(&self, region: Region, has_failed_for: u8) {
```

Comment:
Maybe unified the name of `has_failed_for` and `failure_count` and `failed_for`.

BTW can we have a better name of `retry_start_observe`
`start_observe` -> `retry_start_observe` -> `try_start_observe` -> `retry_observe` seems a litter confuse.